### ISSUE TITLE
0.65.0 Style/TrailingCommaInArguments wants to add comma after heredoc

### ISSUE THREAD

**Issue by jpaulgs** [rubocop-hq/rubocop#6837]
I've got a scenario where the test `Style/TrailingCommaInArguments` wants to add a comma after the start of a heredoc.  This is not the same as #6629

Given the following ruby:
```ruby
class CommaExample
  def self.prefix(first, second)
    puts "#{first}, #{second}"
  end

  def self.message
    prefix('This is',
           <<~MSG
             a message
           MSG
          )
  end
end
```

running rubocop returns 

```bash
rubocop comma_example.rb
Inspecting 1 file
C

Offenses:

comma_example.rb:8:12: C: Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call.
           <<~MSG
           ^^^^^^

1 file inspected, 1 offense detected
```

--------

## Expected behavior

I would not expect this style violation to trigger!

## Actual behavior

Rubocop expects a ',' after the message e.g. `<<~MSG,`

```ruby
  def self.message
    prefix('This is',
           <<~MSG,
             a message
           MSG
          )
  end
```

## Steps to reproduce the problem

See the example code above

## RuboCop version

```
rubocop --version
0.65.0
```

**Comment by CUnknown** [comment]
Same goes for plain SQL block:

<<-SQL
*snip*
      SQL

**Comment by tejasbubane** [comment]
Related to https://github.com/rubocop-hq/rubocop/issues/6701 cc @abrom

**Comment by abrom** [comment]
Not sure why you believe this is an error!? The message is:
```
Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call.
```
It's saying your configuration expects to see a comma after the last parameter. The heredoc is the last parameter, thus there should be a comma. 

I'm guessing you've overridden the `EnforcedStyleForMultiline` configuration to either `comma` or `consistent_comma` ??

https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/TrailingCommaInArguments

**Comment by tejasbubane** [comment]
@abrom Should it not put the comma `after` the heredoc? (i.e. it should correctly recognize where heredoc ends)

### Actual

```ruby
    prefix('This is',
           <<~MSG,
             a message
           MSG
          )
```

### Expected

```ruby
    prefix('This is',
           <<~MSG
             a message
           MSG,
          )
```

**Comment by abrom** [comment]
No, your expected code block is invalid Ruby. The closing heredoc tag must be the only thing on the line. The comma should be after `<<~MSG` as Rubocop is suggesting (given the supposed configuration and provided code example).

**Comment by nhippenmeyer** [comment]
I'm experiencing this issue as well.

```ruby
<<~SQL
  SELECT *
  FROM web_ingress.locations
  ORDER BY id
  LIMIT 10000
  OFFSET 0;
SQL
```

is being corrected to the following (which is invalid SQL)

```ruby
<<~SQL,
  SELECT *
  FROM web_ingress.locations
  ORDER BY id
  LIMIT 10000
  OFFSET 0;
SQL
```

**Comment by koic** [comment]
@nhippenmeyer I cannot reproduce that issue with the latest RuboCop (1.10.0). If the latest RuboCop can reproduce it, can you open a new issue?

**Comment by abrom** [comment]
the trailing comma won't be included in the HEREDOC SQL string @nhippenmeyer. What of the SQL is invalid?

**Comment by nhippenmeyer** [comment]
@koic I'm running RuboCop v1.10.0 so will open a new issue. 

@abrom I misspoke. The problem isn't that the generated SQL string is invalid, but rather the trailing comma on the first line of the heredoc produces a ruby syntax error.

Running the following `query.rb` file results in the error: `query.rb:1: syntax error, unexpected end-of-input`

```ruby
query=<<~SQL,
SELECT *
FROM web_ingress.locations
ORDER BY id
LIMIT 10000
OFFSET 0;
SQL
```

**Comment by nhippenmeyer** [comment]
@koic I'm running RuboCop v1.10.0 so will open a new issue for this.

@abrom I misspoke. The problem is not that the generated SQL string is invalid, but rather the trailing comma on the first line of the heredoc produces a ruby syntax error.

Running the following `query.rb` file

```ruby
def query(limit)
  <<~SQL,
    SELECT *
    FROM web_ingress.locations
    ORDER BY id
    LIMIT #{limit}
    OFFSET 0;
  SQL
end

puts query(0)
```

Results in the error:

```bash
query.rb:2: syntax error, unexpected ',', expecting `end'
  <<~SQL,
query.rb:11: syntax error, unexpected end-of-input, expecting `end'
```

**Comment by nhippenmeyer** [comment]
I'm confirmed this is no longer an issue in the latest version of RuboCop. Apologies for the false alarm.
